Assets/Scripts/GameService.cs

```diff
@@ -53,6 +53,11 @@ private void Start()
     {
         // Initialize all Services.
         soundService = new SoundService(soundScriptableObject, audioEffectSource, backgroundMusicSource);
+    }
+
+    public void Initialize()
+    {
+        SetDifficultyVariables();
         playerService = new PlayerService(playerPrefab, playerScriptableObject, playerBulletPrefab, playerBulletScriptableObject);
```

Comment:
All Services will be initialized in Start() no need for Initialize() or SetDifficultyVariables() as it is not GameService's responsibility.